core/src/main/java/de/danoeh/antennapod/core/storage/DBWriter.java

```diff
@@ -73,6 +74,57 @@ private DBWriter() {
     }
 
     /**
+     * Moves a downloaded FeedMedia file to the archives directory.
+     *
+     * @param context A context that is used for opening a database connection.
+     * @param mediaId ID of the FeedMedia object whose downloaded file should be archived.
+     */
+    public static Future<?> archiveFeedMediaOfItem(final Context context,
+                                                  final long mediaId) {
+        return dbExec.submit(() -> {
+            final FeedMedia media = DBReader.getFeedMedia(mediaId);
+            if (media != null) {
+                Log.i(TAG, String.format("Requested to archive FeedMedia [id=%d, title=%s, downloaded=%s",
+                        media.getId(), media.getEpisodeTitle(), String.valueOf(media.isDownloaded())));
+                boolean result = false;
+                if (media.isDownloaded()) {
+                    // archive downloaded media file
+                    File mediaFile = new File(media.getFile_url());
+                    if (mediaFile.exists()) {
+                        File dest = getNonExistingFile(UserPreferences.getArchiveFolder(media.getItem().getFeed().getTitle())
+                                + File.separator + mediaFile.getName());
+                        Log.i(TAG, "Archiving file to: " + dest);
+                        result = mediaFile.renameTo(dest);
```

Comment:
This won't work with external storage. "Some of the more likely failures include: Both paths be on the same mount point" [1].

--
[1] http://developer.android.com/intl/in/reference/java/io/File.html#renameTo(java.io.File)